sinks/cortex/cortex.go

```diff
@@ -227,10 +227,10 @@ func (s *CortexMetricSink) Flush(ctx context.Context, metrics []samplers.InterMe
 	}
 
 	var batch []samplers.InterMetric
-	for i, metric := range metrics {
+	for _, metric := range metrics {
 		err := doIfNotDone(func() error {
 			batch = append(batch, metric)
-			if i > 0 && i%s.batchWriteSize == 0 {
+			if len(batch)%s.batchWriteSize == 0 {
```

Comment:
For a flush w/ a single batch, will this logic change not cause duplicate flush due to this subsequent logic: https://github.com/stripe/veneur/blob/5ba47fc4a5db53f1edaf7cff3bf21dae81409107/sinks/cortex/cortex.go#L252-L264

Reply:
This logic handles a "leftover" batch. The main loop is meant to consume as many "batchWriteSize" batches as possible, with this picking up the remainder. I added a test case to assert that this is the behavior when "batchWriteSize==numMetrics" (only writes metrics once)